routes/api/v1/favorites.js

```diff
@@ -0,0 +1,31 @@
+var express = require('express');
+var favoritesRouter = express.Router();
+var pry = require('pryjs');
+var User = require('../../../models').User;
+var Location = require('../../../models').Location;
+var UserLocation = require('../../../models').UserLocation;
+var fetch = require('node-fetch');
+// eval(pry.it);
+
+/* POST favorite listing. */
+favoritesRouter.post("/", function(req, res, next) {
+  User.findOne({ where: { api_key: req.body.api_key } })
+  .then(user => {
+    if (user === null) {
+      res.setHeader("Content-Type", "application/json");
+      res.status(401).send(JSON.stringify("Invalid API Key"))
+    } else {
+      var location = req.body.location
+      Location.findOrCreate({ where: {name: location} })
+      .then(([location, created]) => {
+        UserLocation.findOrCreate({ where: {UserId: user.id, LocationId: location.id} })
+        .then(([userLocation, created]) => {
+          res.setHeader("Content-Type", "application/json");
+          res.status(201).send(JSON.stringify({"message":`${location.name} has been added to your favorites`}));
+        })
+      })
+    };
```

Comment:
On this else condition, you might be able to refactor in to a separate function pretty easily.

Additionally, you can return the database queries promises to prevent the need to chain indent them like you're having to right now. Not sure if it's actually better, but I think it makes it easier to follow. Something like this might work:

``` javascript
} else {
   return Location.findOrCreate({ where: {name: location} })
}
.then(([location, created]) => {
   location = location.name // would need to be declared at the top, before the user lookup
   return UserLocation.findOrCreate({ where: {UserId: user.id, LocationId: location.id} })
}
.then(([userLocation, created]) => {
   res.setHeader("Content-Type", "application/json");
   res.status(201).send(JSON.stringify({"message":`${location.name} has been added to your favorites`}));
}
``` 

Lastly, a catch block just to cover the possibility that one of the database creates fail would be an easy add to the end just in case something goes wrong.

Overall looks good.

Reply:
Thank you @plapicola! That makes total sense, and I'll make those changes now.